### PR TITLE
Adds auxcov Code Coverage library

### PR DESCRIPTION
**Pull request by Cyberboss**
- Splits the debug_server crate into a new instruction_hooking crate.
- Adds auxcov crate which uses instruction_hooking to provide line by line code coverage. (GPL)
- Adds setting for rust-anaylzer which properly sets the target to i

Working concept can be viewed here: https://app.codecov.io/gh/tgstation/tgstation

### CODE REVIEW

**Comment by Cyberboss** [comment]
Please see #66 for the build fixes

File: README.md [hunk 1]
```diff
@@ -85,5 +87,5 @@ cargo build --release --target i686-pc-windows-msvc
 Auxtools is licensed under the [MIT license](https://en.wikipedia.org/wiki/MIT_License).
 See [LICENSE](./LICENSE) for more details.
 
-The Auxtool Debug Server (located @ ./debug_server) is licensed under the [GPL 3.0 license](https://www.gnu.org/licenses/gpl-3.0.en.html).
-See [debug_server/LICENSE](./debug_server/LICENSE) for more details.
+The Auxtool Debug Server (located @ ./debug_server) and Auxcov Code Coverage tool (located @ ./auxcov) is licensed under the [GPL 3.0 license](https://www.gnu.org/licenses/gpl-3.0.en.html).
+See [debug_server/LICENSE](./debug_server/LICENSE) or [auxcov/LICENSE](./auxcov/LICENSE) for more details.
```
Comment (willox) [inline README.md:91]: Can we clarify the licensing of instruction_hooking here too

File: auxtools/src/proc.rs [hunk 2]
```diff
@@ -66,6 +66,23 @@ impl Proc {
...
+			return None;
+		}
+
+		let file_id = raw_types::strings::StringId(bytecode[0x01]);
+		if !file_id.valid() {
+			return None;
+		}
+
+		let mut entry: *mut raw_types::strings::StringEntry = std::ptr::null_mut();
+		assert_eq!(raw_types::funcs::get_string_table_entry(&mut entry, file_id), 1);
```
Comment (willox) [inline auxtools/src/proc.rs:81]: These two lines don't seem to be necessary

File: instruction_hooking/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		hooks.borrow_mut().clear());
+}
+
+// Handles any instruction BYOND tries to execute.
+// This function has to leave `*CURRENT_EXECUTION_CONTEXT` in EAX, so make sure to return it.
+#[no_mangle]
+extern "C" fn handle_instruction(
+	ctx: *mut raw_types::procs::ExecutionContext,
+) -> *const raw_types::procs::ExecutionContext {
+	INSTRUCTION_HOOKS.with(|hooks_cell|{
```
Comment (willox) [inline instruction_hooking/src/lib.rs:96]: is fast 👉👈? debug server already increases tg init time a lot

Comment (Cyberboss) [reply instruction_hooking/src/lib.rs:96]: Debug server with:
Current spaceman extension: `Initializations complete within 97.6 seconds!`
This version: `Initializations complete within 103.3 seconds!`

Not great i.g. might just make it bloody unsafe again.

**Pull request merged**